lib/ansible/modules/remote_management/redfish/redfish_command.py

```diff
@@ -0,0 +1,252 @@
+    command_list = module.params['command']
+
+    # admin credentials used for authentication
+    creds = {'user': module.params['user'],
+             'pswd': module.params['password']}
+
+    # user to add/modify/delete
+    user = {'userid': module.params['userid'],
+            'username': module.params['username'],
+            'userpswd': module.params['userpswd'],
+            'userrole': module.params['userrole']}
+
+    # Build root URI
+    root_uri = "https://" + module.params['baseuri']
+    rf_uri = "/redfish/v1"
+    rf_utils = RedfishUtils(creds, root_uri)
+
+    # Check that Category is valid
+    if category not in CATEGORY_COMMANDS_ALL:
+        module.fail_json(msg="Invalid Category: %s" % category)
+
+    # Check that all commands are valid
+    for cmd in command_list:
+        # Fail if even one command given is invalid
+        if cmd not in CATEGORY_COMMANDS_ALL[category]:
+            module.fail_json(msg="Invalid Command: %s" % cmd)
+
+    # Organize by Categories / Commands
+    if category == "Accounts":
+        ACCOUNTS_COMMANDS = {
+            "AddUser": rf_utils.add_user,
+            "EnableUser": rf_utils.enable_user,
+            "DeleteUser": rf_utils.delete_user,
+            "DisableUser": rf_utils.disable_user,
+            "UpdateUserRole": rf_utils.update_user_role,
+            "UpdateUserPassword": rf_utils.update_user_password
+        }
+
+        # execute only if we find an Account service resource
+        result = rf_utils._find_accountservice_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        for command in command_list:
+            result = ACCOUNTS_COMMANDS[command](user)
+
+    elif category == "Systems":
+        # execute only if we find a System resource
+        result = rf_utils._find_systems_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
+
+        for command in command_list:
+            if "Power" in command:
+                result = rf_utils.manage_system_power(command)
+            elif command == "SetOneTimeBoot":
+                result = rf_utils.set_one_time_boot_device(module.params['bootdevice'])
+            elif command == "CreateBiosConfigJob":
+                # execute only if we find a Managers resource
+                result = rf_utils._find_managers_resource(rf_uri)
+                if result['ret'] is False:
+                    module.fail_json(msg=result['msg'])
+                result = rf_utils.create_bios_config_job()
+
+    elif category == "Manager":
+        MANAGER_COMMANDS = {
+            "GracefulRestart": rf_utils.restart_manager_gracefully,
+            "ClearLogs": rf_utils.clear_logs
+        }
+
+        # execute only if we find a Manager service resource
+        result = rf_utils._find_managers_resource(rf_uri)
+        if result['ret'] is False:
+            module.fail_json(msg=result['msg'])
```

Comment:
The output of result['msg'] varies in case of error, it returns an HTTPError or URLError code based on different scenarios (in older OOB controllers some APIs are not available --> 404).